src/libs/pddl_parser/tests/test_pddl_parser.cpp

```diff
@@ -0,0 +1,261 @@
+	(:action test-action                        \n\
+	 :parameters (?t - obj-b)                   \n\
+	 :precondition (pred ?t)                    \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)",
+	                   "(define (domain action-unknown-type-in-constant) \n\
+	(:requirements :strips :typing)             \n\
+	(:types                                     \n\
+	 	obj-a - object                            \n\
+	 )                                          \n\
+	(:constants                                 \n\
+	 	TEST_CONST - obj-b                        \n\
+	 )                                          \n\
+	(:predicates                                \n\
+	 	(pred ?r - obj-a)                         \n\
+	 )                                          \n\
+	(:action test-action                        \n\
+	 :parameters (?t - obj-a)                   \n\
+	 :precondition (pred ?t)                    \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)",
+	                   "(define (domain action-constant-missmatch)       \n\
+	(:requirements :strips :typing)             \n\
+	(:types                                     \n\
+	 	obj-a - object                            \n\
+	 	obj-b - object                            \n\
+	 )                                          \n\
+	(:constants                                 \n\
+	 	TEST_CONST - obj-b                        \n\
+	 )                                          \n\
+	(:predicates                                \n\
+	 	(pred ?r - obj-a)                         \n\
+	 )                                          \n\
+	(:action test-action                        \n\
+	 :parameters (?t - obj-a)                   \n\
+	 :precondition (pred TEST_CONST)            \n\
+	 :effect (not (pred ?t))                    \n\
+	 )                                          \n\
+)"};
+	for (const auto &s : benchmarks) {
+		try {
+			PddlParser p;
+			p.parseDomain(s);
+			FAIL() << s << "\n expected type error";
+		} catch (PddlParserException &e) {
+			if (e.error_type != PddlErrorType::TYPE_ERROR) {
+				FAIL() << s << "\n failed with error " << e.what();
+			} else {
+			}
+		}
+	}
+	SUCCEED() << " got expected type errors ";
+}
+
+TEST(PddlParserTest, MinimalDomain)
+{
+	try {
+		PddlParser p;
+		p.parseDomain("(define (domain test-domain)              \n\
+	(:requirements)                  \n\
+	(:predicates                     \n\
+		(pred)                         \n\
+	 )                               \n\
+	(:action test-action             \n\
+	 :parameters (?t)                \n\
+	 :precondition (pred)            \n\
+	 :effect (not (pred))            \n\
+	 )                               \n\
+)");
+		SUCCEED() << "Minimal domain parsed ";
+	} catch (PddlParserException const &e) {
+		FAIL() << " Unexpected Exception: " << e.what();
+	}
+}
```

Comment:
You can use [`EXPECT_NO_THROW`](http://google.github.io/googletest/reference/assertions.html#EXPECT_NO_THROW)